### PR TITLE
doc for NVIDIA EH features

### PR DESCRIPTION
**Pull request by pagezyhf**
Added documentation for:
- Serverless Inference powered by NVIDIA
- Train on NVIDIA DGX Cloud

Some TBC:
- Pricing for Serverless Inference
- Check Nvidia Collection is public

Wait before 10:30 pm 29th July before merging.

### CODE REVIEW

File: docs/hub/inference-nim.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+## Pricing
+
+Usage of Serverless Inference powered by NVIDIA is billed based on the compute time spent per request. Usage fees accrue to your Enterprise Hub Organizations’ current monthly billing cycle, once a job is completed. You can check your current and past usage at any time within the billing settings of your Enterprise Hub Organization.
+
+The total cost for a request will depend on the model size, the number of GPUs required, and the time taken to process the request.
+
+| NVIDIA GPU                | GPU Memory                | On-Demand Price/hr        |
+|---------------------------|---------------------------|---------------------------|
+| NVIDIA L40S               | 48GB                      |$2.75                      |
+| NVIDIA H100               | 80GB                      |$8.25                      |
```
Comment (jeffboudier) [inline docs/hub/inference-nim.md:22]:
```suggestion
| NVIDIA H100               | 80GB                      |$8.25                      |
```

File: docs/hub/_toctree.yml [hunk 2]
```diff
@@ -313,6 +313,10 @@
         title: Dataset viewer for Private datasets
       - local: enterprise-hub-resource-groups
         title: Resource Groups (Advanced Access Control)
+      - local: train-dgx-cloud
+        title: Train on NVIDIA DGX Cloud
+      - local: inference-nim
+        title: Serverless Inference powered by NVIDIA
```
[the remark this answers is not in the document]
Comment (julien-c) [reply docs/hub/_toctree.yml:319]: maybe in the autotrain doc rather

[the remark this answers is not in the document]
Comment (jeffboudier) [reply docs/hub/_toctree.yml:319]: I see what you mean ("external" features?), however they are benefits of the Enterprise Hub subscription (you need to subscribe to EH to use them), so it feels like a good spot for them. Maybe we can group a section "Advanced compute options"?

[the remark this answers is not in the document]
Comment (pagezyhf) [reply docs/hub/_toctree.yml:319]: Another name proposition a bit more generic for the section grouping this kind of features "third-party integrations" .

"A subscription to the Enterprise Hub gives you access to Hugging Face selected third party integrations, built to accelerate the machine learning journey of your organization."

[the remark this answers is not in the document]
Comment (julien-c) [reply docs/hub/_toctree.yml:319]: Ok then let's nest under `Advanced Compute Options ` to be consistent with https://huggingface.co/enterprise (cc @gary149)

**Review by julien-c — APPROVED** [review]

File: docs/hub/advanced-compute-options.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+Serverless Inference powered by NVIDIA offers access to NVIDIA AI technology GPUs an [NVIDIA Inference Microservices](https://www.nvidia.com/en-us/ai/) (NIM) in a serverless way. Use standardized APIs and a few lines of code to run inference in a pay-as-you-go pricing model.
+
+### How it works
+
+Read the [blogpost for Serverless Inference with Hugging Face and NVIDIA NIMs](https://huggingface.co/blog/inference-dgx-cloud#how-it-works).
+
+### Supported models
+
+You can find all supported models in [this NVIDIA Collection](https://huggingface.co/collections/nvidia/nim-66a3c6fcdcb5bbc6e975b508).
```
Comment (pcuenca) [inline docs/hub/advanced-compute-options.md:56]: I can't see this collection.

File: docs/hub/advanced-compute-options.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+Usage of Train on NVIDIA DGX Cloud is billed by the minute of the GPU instances used during your training jobs. Usage fees accrue to your Enterprise Hub Organizations’ current monthly billing cycle, once a job is completed. You can check your current and past usage at any time within the billing settings of your Enterprise Hub Organization.
+
+| NVIDIA GPU                | GPU Memory                | On-Demand Price/hr        |
+|---------------------------|---------------------------|---------------------------|
+| NVIDIA L40S               | 48GB                      |$2.75                      |
+| NVIDIA H100               | 80GB                      |$8.25                      |
+
+## Serverless Inference powered by NVIDIA
+
+Serverless Inference powered by NVIDIA offers access to NVIDIA AI technology GPUs an [NVIDIA Inference Microservices](https://www.nvidia.com/en-us/ai/) (NIM) in a serverless way. Use standardized APIs and a few lines of code to run inference in a pay-as-you-go pricing model.
```
Comment (pcuenca) [inline docs/hub/advanced-compute-options.md:48]:
```suggestion
Serverless Inference powered by NVIDIA offers access to NVIDIA AI technology GPUs on [NVIDIA Inference Microservices](https://www.nvidia.com/en-us/ai/) (NIM) in a serverless way. Use standardized APIs and a few lines of code to run inference in a pay-as-you-go pricing model.
```

The line is a bit long, not sure if `on` is correct but `an` didn't sound right either.

File: docs/hub/advanced-compute-options.md (hunk 3 above)
Comment (jeffboudier) [reply docs/hub/advanced-compute-options.md:56]: Thanks for spotting! Asked NVIDIA to make it public, standby...

File: docs/hub/advanced-compute-options.md (hunk 4 above)
Comment (jeffboudier) [reply docs/hub/advanced-compute-options.md:48]:
```suggestion
Serverless Inference powered by NVIDIA offers access to [NVIDIA Inference Microservices (NIM)](https://www.nvidia.com/en-us/ai/) powered by NVIDIA H100s in a serverless way. Use standardized APIs and a few lines of code to run inference in a pay-as-you-go pricing model.
```
Agreed, trying to clarify

Comment (pagezyhf) [reply docs/hub/advanced-compute-options.md:48]: thank you for the feedback

File: docs/hub/advanced-compute-options.md (hunk 3 above)
Comment (pagezyhf) [reply docs/hub/advanced-compute-options.md:56]: I'll wait for the collection to be public before merging. Thanks for spotting Pedro and Jeff for the follow up

**Pull request merged**